ghw/libghw.c

```diff
@@ -1653,33 +1662,43 @@ ghw_read_cycle_cont (struct ghw_handler *h, int *list)
 	  /* Fast version. */
 	  i = i + d;
 	  if (i >= h->nbr_sigs)
-	    goto err;
+	    goto err; 
 	}
       else
 	{
-	  /* Slow version: Linear search through all signals. Find d-th
-	     element with non-NULL type. Note: Type of sigs[0] is ignored. */
-	  while (d > 0)
-	    {
-	      i++;
-	      if (i >= h->nbr_sigs)
-		goto err;
-	      if (h->sigs[i].type != NULL)
-		d--;
+	  /* build the cache if not existing yet */
+	  if(h->no_null_sig_cache == NULL) {
+	    h->no_null_sig_cache = malloc_unwrap(sizeof(uint32_t)*h->nbr_sigs);
+	    memset(h->no_null_sig_cache,0xFF,sizeof(uint32_t)*h->nbr_sigs);
```

Comment:
Looks ok for me.
I was about to merge it, but re-reading I don't understand why it is initialized to 0xff ?
Later, (line 1689), the value read from the cache is compared to 0.